x-pack/plugins/endpoint/public/applications/endpoint/store/policy_list/middleware.ts

```diff
@@ -26,20 +29,29 @@ export const policyListMiddlewareFactory: MiddlewareFactory<PolicyListState> = c
         pageIndex = state.pageIndex;
       }
 
-      // Need load data from API and remove fake data below
-      // Refactor tracked via: https://github.com/elastic/endpoint-app-team/issues/150
-      const { getFakeDatasourceApiResponse } = await import('./fake_data');
-      const { items: policyItems, total } = await getFakeDatasourceApiResponse(pageIndex, pageSize);
+      try {
+        const { items: policyItems, total } = await sendGetEndpoingDatasources(http, {
+          query: {
+            perPage: pageSize,
+            page: pageIndex + 1,
+          },
+        });
 
-      dispatch({
-        type: 'serverReturnedPolicyListData',
-        payload: {
-          policyItems,
-          pageIndex,
-          pageSize,
-          total,
-        },
-      });
+        dispatch({
```

Comment:
would you consider moving the `dispatch` call outside the `try` block? If the reducer throws a runtime error, it'll be treated the same as if the HTTP call returned an error code.

Reply:
ahh - yes. good point. I personally like to keep the code inside of `try {}` blocks to a minimum since (I think) the JS compilers are not able to optimize that code. will move it out.